jobs/src/main/scala/dcos/metronome/model/JobSpec.scala

```diff
@@ -29,14 +33,36 @@ case class JobSpec(
 
 object JobSpec {
   val DefaultDescription = None
+  val DefaultDependencies = Option.empty[Seq[JobId]]
   val DefaultLabels = Map.empty[String, String]
   val DefaultSchedule = Seq.empty[ScheduleSpec]
   val DefaultRunSpec = JobRunSpec()
 
-  implicit lazy val validJobSpec: Validator[JobSpec] = validator[JobSpec] { jobSpec =>
+  import com.wix.accord.ViolationBuilder._
+
+  private def unique[T]: Validator[Seq[T]] = new NullSafeValidator[Seq[T]](
+    test = col => col.size == col.toSet.size,
+    failure = _ -> s"has double entries"
+  )
+
+  def acyclic[JobId]: Validator[Seq[JobId]] = new NullSafeValidator[Seq[JobId]](
+    test = _ => false, // TODO: check if the dependencies are cyclic.
+    failure = _ -> "is not acyclic"
+  )
+
+  def validJobSpec(jobSpecService: JobSpecService): Validator[JobSpec] = validator[JobSpec] { jobSpec =>
+
+    def exist: Validator[Seq[JobId]] = new NullSafeValidator[Seq[JobId]](
+      test = ids => ids.exists(id => Await.result(jobSpecService.getJobSpec(id), ???).isEmpty),
```

Comment:
I really do not want to block here. So I think we should move this into the controller.